hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/rollback/BaseRollbackActionExecutor.java

```diff
@@ -297,4 +301,35 @@ protected void dropBootstrapIndexIfNeeded(HoodieInstant instantToRollback) {
       BootstrapIndex.getBootstrapIndex(table.getMetaClient()).dropIndex();
     }
   }
+
+  private void backupRollbackInstantsIfNeeded() {
+    if (!config.shouldBackupRollbacks()) {
+      // Backup not required
+      return;
+    }
+
+    Path backupDir = new Path(config.getRollbackBackupDirectory());
+    if (!backupDir.isAbsolute()) {
+      // Path specified is relative to the meta directory
+      backupDir = new Path(table.getMetaClient().getMetaPath(), config.getRollbackBackupDirectory());
+    }
+
+    // Determine the instants to backup. Completed and inflight instants are backed up.
+    HoodieActiveTimeline activeTimeline = table.getActiveTimeline();
+    List<HoodieInstant> instantsToBackup = new ArrayList<HoodieInstant>(3);
+    instantsToBackup.add(instantToRollback);
+    if (instantToRollback.isCompleted()) {
+      instantsToBackup.add(HoodieTimeline.getInflightInstant(instantToRollback, table.getMetaClient()));
```

Comment:
lets ensure we backup all 3 if present. (requested, inflight and completed instant)